Loop/Managers/DeviceDataManager.swift

```diff
@@ -697,8 +789,9 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
             if let sentrySupported = pumpState?.pumpModel?.larger where !sentrySupported {
                 rileyLinkManager.idleListeningEnabled = false
             }
-
             NSUserDefaults.standardUserDefaults().pumpModelNumber = pumpState?.pumpModel?.rawValue
+            self.enableRileyLinkHeartbeatIfNeeded()
```

Comment:
If they are using share and switch from a 523 pump to a 522 pump, then don't we need to enable the heartbeat?